src/components/App.js

```diff
@@ -32,6 +35,21 @@ class App extends Component {
                                 <StartPage data={this.props.data}/>
                             </ErrorBoundary>
                         </Route>
+                        <Route path="/add">
+                            <ErrorBoundary>
+                                <AddPage />
+                            </ErrorBoundary>
+                        </Route>
+                        <Route path="/edit">
+                            <ErrorBoundary>
+                                <EditPage />
+                            </ErrorBoundary>
+                        </Route>
+                        <Route path="/delete/:id">
+                            <ErrorBoundary>
+                                <Popup/>
+                            </ErrorBoundary>
+                        </Route>
```

Comment:
According to the task, all of these should be modal windows:

![image](https://user-images.githubusercontent.com/31325372/91957134-51726d00-ed0e-11ea-92d2-a94f505f5550.png)

But the way those could still be loaded with a route, but you shouldn't destroy the main layout.